third_party/libunwindstack/PeCoffInterface.cpp

```diff
@@ -236,6 +242,12 @@ bool PeCoffInterface<AddressType>::ParseOptionalHeader(uint64_t offset) {
     }
   }
 
+  if (coff_memory_.cur_offset() != end_offset) {
```

Comment:
What about doing this check before the for-loop above?

If the file is really invalid we could avoid trying to allocate an arbitrary amount of data in the `optional_header_.data_dirs.resize(optional_header_.num_data_dir_entries);` call and potentially
crash the application which leads to bad user experience.

I guess `optional_header_.num_data_dir_entries` should be identical to `(end_offset - coff_memory_.cur_offset()) / 64` when checked before the for-loop.

Reply:
(Of course that doesn't really help when `coff_header_.hdrsize` is also wrong.)

Reply:
Yes, good point about allocating based on num_data_dir_entries. Will change based on your suggestion!

Reply:
I'll also add tests with inconsistent data to check if we fail to initialize.

Reply:
Ok, should be better now. We still have the case where `hdrsize` and `num_dir_data_entries` are both wrong but consistent with each other. While `num_dir_data_entries` can be fairly large (causing a large allocation), it can't be fairly large _and_ consistent with hdrsize, which is only a uint16.

Reply:
Awesome. Even better.